src/umockcallrecorder.c

```diff
@@ -829,3 +829,23 @@ int umockcallrecorder_can_call_fail(UMOCKCALLRECORDER_HANDLE umock_call_recorder
 
     return result;
 }
+
+char* umockcallrecorder_get_call_description(UMOCKCALLRECORDER_HANDLE umock_call_recorder, size_t index)
+{
+    char* result = NULL;
+
+    /* Codes_SRS_UMOCKCALLRECORDER_31_056: [ If umock_call_recorder or call_can_fail is NULL, umockcallrecorder_can_call_fail shall return a non-zero value. ]*/
+    if ((umock_call_recorder == NULL) ||
+        /* Codes_SRS_UMOCKCALLRECORDER_31_057: [ If index is greater or equal to the current expected calls count, umockcallrecorder_can_call_fail shall return a non-zero value. ]*/
+        (index >= umock_call_recorder->expected_call_count))
+    {
```

Comment:
Same here, missing .md specs update, which should be used to write also the tests for it.

Reply:
Addressed most of the issues. I'm having this error though, need to understand why.

```shell
[ 93%] Linking C executable ../../../Debug/umock_c_negt_noint_tests_exe_umock_c/umock_c_negt_noint_tests_exe_umock_c
/usr/bin/ld: ../../../libumock_c.a(umockcallrecorder.c.o): in function `umockcallrecorder_destroy':
/home/ewertons/code/s2/umock-c/src/umockcallrecorder.c:150: multiple definition of `umockcallrecorder_destroy'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:155: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockcallrecorder.c.o): in function `umockcallrecorder_clone':
/home/ewertons/code/s2/umock-c/src/umockcallrecorder.c:595: multiple definition of `umockcallrecorder_clone'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:93: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockcallrecorder.c.o): in function `umockcallrecorder_get_expected_call_count':
/home/ewertons/code/s2/umock-c/src/umockcallrecorder.c:722: multiple definition of `umockcallrecorder_get_expected_call_count'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:106: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockcallrecorder.c.o): in function `umockcallrecorder_fail_call':
/home/ewertons/code/s2/umock-c/src/umockcallrecorder.c:752: multiple definition of `umockcallrecorder_fail_call'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:120: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockcallrecorder.c.o): in function `umockcallrecorder_can_call_fail':
/home/ewertons/code/s2/umock-c/src/umockcallrecorder.c:791: multiple definition of `umockcallrecorder_can_call_fail'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:147: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockalloc.c.o): in function `umockalloc_malloc':
/home/ewertons/code/s2/umock-c/src/umockalloc.c:10: multiple definition of `umockalloc_malloc'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:229: first defined here
/usr/bin/ld: ../../../libumock_c.a(umockalloc.c.o): in function `umockalloc_free':
/home/ewertons/code/s2/umock-c/src/umockalloc.c:31: multiple definition of `umockalloc_free'; libumock_c_negt_noint_tests_lib_umock_c.a(umock_c_negt_noint_tests.c.o):/home/ewertons/code/s2/umock-c/tests/umock_c_negt_noini_ut/umock_c_negt_noint_tests.c:234: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [tests/umock_c_negt_noini_ut/exe/CMakeFiles/umock_c_negt_noint_tests_exe_umock_c.dir/build.make:104: Debug/umock_c_negt_noint_tests_exe_umock_c/umock_c_negt_noint_tests_exe_umock_c] Error 1
make[1]: *** [CMakeFiles/Makefile2:2657: tests/umock_c_negt_noini_ut/exe/CMakeFiles/umock_c_negt_noint_tests_exe_umock_c.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Apparently the build is not happy that I introduced a new function in umockcallrecorder.c

Reply:
Note: I took the dependency on this new function off the PR I was actually working on so it gets unblocked. I'll come back to this once I get the other task done.